src/registrar/tests/test_models_domain.py

```diff
@@ -705,6 +705,123 @@ def test_updates_security_email(self):
         self.mockedSendFunction.assert_has_calls(expected_calls, any_order=True)
         self.assertEqual(PublicContact.objects.filter(domain=self.domain).count(), 1)
 
+    def test_not_disclosed_on_other_contacts(self):
+        """
+        Scenario: Registrant creates a new domain with multiple contacts
+            When `domain` has registrant, admin, technical,
+                and security contacts
+            Then Domain sends `commands.CreateContact` to the registry
+            And the field `disclose` is set to false for DF.EMAIL
+                on all fields except security
+        """
+        # Generates a domain with four existing contacts
+        domain, _ = Domain.objects.get_or_create(name="freeman.gov")
+
+        # Contact setup
+        expected_admin = domain.get_default_administrative_contact()
+        expected_admin.email = self.mockAdministrativeContact.email
+
+        expected_registrant = domain.get_default_registrant_contact()
+        expected_registrant.email = self.mockRegistrantContact.email
+
+        expected_security = domain.get_default_security_contact()
+        expected_security.email = self.mockSecurityContact.email
+
+        expected_tech = domain.get_default_technical_contact()
+        expected_tech.email = self.mockTechnicalContact.email
+
+        domain.administrative_contact = expected_admin
+        domain.registrant_contact = expected_registrant
+        domain.security_contact = expected_security
+        domain.technical_contact = expected_tech
+
+        contacts = [
+            (expected_admin, domain.administrative_contact),
+            (expected_registrant, domain.registrant_contact),
+            (expected_security, domain.security_contact),
+            (expected_tech, domain.technical_contact),
+        ]
+
+        # Test for each contact
+        for contact in contacts:
+            expected_contact = contact[0]
+            actual_contact = contact[1]
+            is_security = actual_contact.contact_type == "security"
+
+            expectedCreateCommand = self._convertPublicContactToEpp(
```

Comment:
In general, here we are just copying the code here we are trying to test. That violates the basic point of our test. There is no point to this test if we are just testing a function against itself. I would manually write out the call we expect like in the other tests.

Reply:
Looks like there is a typo here on is_security. Should be testing on expected_contact.contact_type.

Anywho, I have no problem writing this out manually but in general, I am a bit confused as to how this is always going to return true. Since this is a helper function in common.py, that code is completely decoupled from the disclose function. It doesn't use _convertPublicContactToEpp from domain.py at all, its using the one in common.py
![image](https://github.com/cisagov/getgov/assets/141044360/9f432faa-c347-4f68-a638-cc0ae1e97060)

![image](https://github.com/cisagov/getgov/assets/141044360/e720d623-41be-48e8-9d48-b84c043cb004)

Reply:
So first, yes I meant this test specifically. And it's not that you are exactly using the same function but it comes down to a coding best practices concept of limiting code logic within a test and if logic exists in a test, it itself should be tested. Here you are fundamentally checking the disclose is correct. However, you are using code to generate what the expected disclose should be without testing that code. Your code to generate the expected disclose is also almost _identical_ to the very disclose code you are testing. 
Look how close these codes are, basically the only difference is one is passing a boolean instead of making it. Left is the code we are testing, right is the code you are using to test the left...
![image](https://github.com/cisagov/getgov/assets/109625347/0650cc01-1c36-4ff4-ae21-c4bc6051d2fc)

Reply:
I just do not view this as falling under the category as a valid test due to the duplication of code and the original intent of this test being to explicitly test the disclose. I already requested changes, but I won't block this again, if you want to merge you may.

Reply:
I understand your concern with how similar the checks are, I think its entirely reasonable to keep tests as simple as possible so that we actually perform a good test, rather than a bad one. In that case, we should be testing the other use cases of _convertPublicContactToEpp() to best conform to that practice, given that we do use it elsewhere. test_no_security_email is a great example because it needs to explicitly check if its not getting disclosed. See below... 

![image](https://github.com/cisagov/getgov/assets/141044360/d2d8126c-3e03-46b1-8b56-429e157f1293)
![image](https://github.com/cisagov/getgov/assets/141044360/5392a383-8dad-4e8a-9cf8-57452a11f096)

Maybe a good middle ground here would be a test case on the _convertPublicContactToEpp() function itself? What do you think?